### PR TITLE
Cache groups_hash

### PR DESCRIPTION
**Pull request by Ana06**
It didn't work in https://github.com/openSUSE/open-build-service/pull/6806 :relaxed:

### CODE REVIEW

File: src/api/app/models/staging/workflow.rb [hunk 1]
```diff
@@ -57,12 +57,13 @@ def self.load_groups
     #
     # TODO: Refactor this code using to_h when updating to Ruby 2.6 (performance improvement)
     #
-    groups_hash = {}
-    # FIXME: try to use cache without failure. Like: Rails.cache.fetch(Group.all.cache_key) do
-    Group.find_each do |group|
-      groups_hash[group.title] = group
+    Rails.cache.fetch(Group.all.cache_key) do
```
Comment (coolo) [inline src/api/app/models/staging/workflow.rb]: you have to make sure no other code is using this cache_key to cache another return value, so better append some code path specific key

File: src/api/app/models/staging/workflow.rb [hunk 2]
```diff
@@ -56,13 +56,13 @@ def self.load_groups
     # Group.where(id: group_ids).each do |group|
     #
     # TODO: Refactor this code using to_h when updating to Ruby 2.6 (performance improvement)
-    #
-    groups_hash = {}
-    # FIXME: try to use cache without failure. Like: Rails.cache.fetch(Group.all.cache_key) do
-    Group.find_each do |group|
-      groups_hash[group.title] = group
+    Rails.cache.fetch("groups_hash_#{Group.all.cache_key}") do
```
Comment (ChrisBr) [inline src/api/app/models/staging/workflow.rb]: Why is the prefix needed?

Comment (Ana06) [reply src/api/app/models/staging/workflow.rb]: [@coolo's comment](https://github.com/openSUSE/open-build-service/pull/6815#discussion_r249755960):

> you have to make sure no other code is using this cache_key to cache another return value, so better append some code path specific key

Comment (Ana06) [reply src/api/app/models/staging/workflow.rb]: I think it doesn't hurt

Comment (ChrisBr) [reply src/api/app/models/staging/workflow.rb]: I'm fine with coolo's explanation.

Comment (coolo) [reply src/api/app/models/staging/workflow.rb]: note that `['groups_hash', Group.all_cache_key] ` seems to be the stackoverflow preferred way to write this :)

**Pull request merged**